test/support/factories/alerts/alert.ex

```diff
@@ -28,4 +31,40 @@ defmodule Test.Support.Factories.Alerts.Alert do
       url: Faker.Internet.url()
     }
   end
+
+  def for_route(alert, route_id) do
+    %{
+      alert
+      | informed_entity:
+          InformedEntitySet.build(:informed_entity_set,
+            route: route_id,
+            entities: [
+              InformedEntity.build(:informed_entity, route: route_id)
+            ]
+          )
+    }
+  end
+
+  def with_high_priority_effect(alert) do
+    %{alert | effect: Faker.Util.pick(@high_priority_effects)}
+  end
```

Comment:
Similarly, this could be

```elixir
def high_priority_alert_factory do
  build(:alert, effect: Faker.Util.pick(@high_priority_effects))
end
```

to enable `high_priority_alerts = build_list(4, :high_priority_alert)`

Reply:
This comment and [this one](https://github.com/mbta/dotcom/pull/2347#discussion_r1935955744) kind of go together to show why I didn't do it that way though, because I wanted to be able to chain them together, like 👇 
```elixir
build(:alert, effect: effect) |> active_during(time) |> for_route(affected_route_id)
```

Reply:
It'd save you a chain or two, e.g. `build(:alert_for_route, route_id: affected_route_id) |> active_during(time)`. But if you like each property being a separate step in the chain that's a choice.... but if you're doing the same steps over and over it might be more useful to make a factory tailored to that use case.

Reply:
Yeah that's fair - I guess I wasn't sure where to draw the line between what should be chained and what should be "top-level".

I was loosely following [this](https://hexdocs.pm/ex_machina/readme.html#flexible-factories-with-pipes)

Reply:
Okay, fair enough!